b3lb/rest/models.py

```diff
@@ -368,3 +398,116 @@ class Meta(object):
+        (LOCK_SETTINGS_DISABLE_MIC, "{} - Setting to 'true' will only allow user to join listen only.".format(LOCK_SETTINGS_DISABLE_MIC)),
+        (LOCK_SETTINGS_DISABLE_PRIVATE_CHAT, "{} - Setting to 'true' will disable private chats in the meeting.".format(LOCK_SETTINGS_DISABLE_PRIVATE_CHAT)),
+        (LOCK_SETTINGS_DISABLE_PUBLIC_CHAT, "{} - Setting to 'true' will disable public chat in the meeting.".format(LOCK_SETTINGS_DISABLE_PUBLIC_CHAT)),
+        (LOCK_SETTINGS_DISABLE_NOTE, "{} - Setting to 'true' will disable notes in the meeting.".format(LOCK_SETTINGS_DISABLE_NOTE)),
+        (LOCK_SETTINGS_LOCKED_LAYOUT, "{} - Setting to 'true' will lock the layout in the meeting.".format(LOCK_SETTINGS_LOCKED_LAYOUT)),
+        (LOCK_SETTINGS_HIDE_USER_LIST, "{} - Setting to 'true' will lock the user list in the meeting.".format(LOCK_SETTINGS_HIDE_USER_LIST)),
+        (LOCK_SETTINGS_LOCK_ON_JOIN, "{} - Setting to 'false' will not apply lock setting to users when they join.".format(LOCK_SETTINGS_LOCK_ON_JOIN)),
+        (LOCK_SETTINGS_LOCK_ON_JOIN_CONFIGURABLE, "{} - Setting to 'true' will allow applying of lockSettingsLockOnJoin param.".format(LOCK_SETTINGS_LOCK_ON_JOIN_CONFIGURABLE)),
+        (GUEST_POLICY, "{} - Will set the guest policy for the meeting. Possible values are ALWAYS_ACCEPT, ALWAYS_DENY, and ASK_MODERATOR.".format(GUEST_POLICY)),
+        (MEETING_KEEP_EVENT, "{} - If set to 'true' BigBlueButton saves meeting events even if the meeting is not recorded.".format(MEETING_KEEP_EVENT))
+    )
+
+    MODE_CHOICES = [
+        (BLOCK, BLOCK),
+        (SET, SET),
+        (OVERRIDE, OVERRIDE)
+    ]
+
+    BOOLEAN_REGEX = r'^(true|false)$'
+    BOOLEAN_PARAMETER = [
+        WEBCAMS_ONLY_FOR_MODERATOR,
+        MUTE_ON_START,
+        ALLOW_MODS_TO_UNMUTE_USERS,
+        LOCK_SETTINGS_DISABLE_CAM,
+        LOCK_SETTINGS_DISABLE_MIC,
+        LOCK_SETTINGS_DISABLE_PRIVATE_CHAT,
+        LOCK_SETTINGS_DISABLE_PUBLIC_CHAT,
+        LOCK_SETTINGS_DISABLE_NOTE,
+        LOCK_SETTINGS_LOCKED_LAYOUT,
+        LOCK_SETTINGS_HIDE_USER_LIST,
+        LOCK_SETTINGS_LOCK_ON_JOIN,
+        LOCK_SETTINGS_LOCK_ON_JOIN_CONFIGURABLE,
+        MEETING_KEEP_EVENT
+    ]
+
+    NUMBER_REGEX = r'^[\d]+$'
+    NUMBER_PARAMETERS = [MAX_PARTICIPANTS, DURATION]
+
+    POLICY_REGEX = r'^(ALWAYS_ACCEPT|ALWAYS_DENY|ASK_MODERATOR)$'
+    POLICY_PARAMETERS = [GUEST_POLICY]
+
+    COLOR_REGEX = r'^#[A-F0-9]{6}$'
+    COLOR_PARAMETER = [BANNER_COLOR]
+
+    mode = models.CharField(max_length=10, choices=MODE_CHOICES)
+    parameter = models.CharField(max_length=64, choices=PARAMETER_CHOICES)
+    tenant = models.ForeignKey(Tenant, on_delete=models.CASCADE)
+    value = models.CharField(max_length=250, blank=True, null=True)
+
+    def clean_fields(self, exclude=None):
+        if self.parameter in self.BOOLEAN_PARAMETER:
```

Comment:
Skip validation if `mode` is `BLOCK`.

Reply:
validation skipped if not `SET` or `OVERRIDE`